src/NUnitFramework/framework/Internal/Execution/CompositeWorkItem.cs

```diff
@@ -231,13 +236,32 @@ private void RunChildren()
 
         private void CreateChildWorkItems()
         {
-            _children = new List<WorkItem>();
+            Children = new List<WorkItem>();
 
             foreach (ITest test in _suite.Tests)
                 if (_childFilter.Pass(test))
-                    _children.Add(WorkItem.CreateWorkItem(test, _childFilter));
+                {
+                    Children.Add(WorkItem.CreateWorkItem(test, _childFilter));
+                    if (test.Properties.ContainsKey(PropertyNames.Order)) CountOrder++;
+                }
+
```

Comment:
Yep, i was looking at that, I didn't realize until i re-read your post that Sort has the ability to do first N items, never used this before, I should RTFM more often :)

Had to create a class for backwards compatibility with .NET 2 as the Create method on Comparer is only available in 4.5, so its a little ugly but working.